pkg/detector/detector.go

```diff
@@ -223,6 +223,10 @@ func (d *ResourceDetector) Reconcile(key util.QueueKey) error {
 		return err
 	}
 
+	if shouldIgnorePolicyMatching(object) {
```

Comment:
Would it be more appropriate to put this logic into `func (d *ResourceDetector) EventFilter?

Reply:
Good idea!

Reply:
Well, I just realized that we need to get the origin object. However, the parameter of `EventFilter` is `ClusterWideKey`.  So we need to retrieve the origin object in `EventFilter`. Maybe it's not a proper solution......

Reply:
The parmater of `EventFilter` is `not` `ClusterWideKey`, it's the object. And we don't need to retrieve the original object from apiserver or cache.

Reply:
If I understand right, this piece of code should work.
```diff
diff --git a/pkg/detector/detector.go b/pkg/detector/detector.go
index 4f0da7ea..51e355e1 100644
--- a/pkg/detector/detector.go
+++ b/pkg/detector/detector.go
@@ -318,6 +318,15 @@ func (d *ResourceDetector) EventFilter(obj interface{}) bool {
                return false
        }

+       // The secret, with type 'kubernetes.io/service-account-token', is created along with `ServiceAccount` should be
+       // prevented from propagating.
+       // Refer to https://github.com/karmada-io/karmada/pull/1525#issuecomment-1091030659 for more details.
+       if st, ok := obj.(corev1.Secret); ok {
+               if st.Type == corev1.SecretTypeServiceAccountToken {
+                       return false
+               }
+       }
+
        return true
 }

```

Reply:
Great, that's true. I made a mistake.

Reply:
> Maybe because we use a dynamic informer manager.

Yes, exactly. I forgot this.